### ISSUE TITLE
Transcoding settings to convert to hevc

### ISSUE THREAD

**Issue by pcxnet** [clinton-hall/nzbToMedia#1756]
I am trying to get my system to convert all non x265/hevc files to hevc only if they are not already.

Eg, if it downloads The.Simpsons.S08E22.1080p.HEVC.x265-MeGusta , it shouldn't do transcoding since its already HEVC/x265 but if it was H264/X264 or anything else it would transcode.

I have the below settings but it still tries to transcode files that are already hevc.

Im sure im doing something wrong, but can't see what.

    #### Define custom settings below.
    outputVideoExtension = .mkv
    outputVideoCodec = hevc_nvenc
    VideoCodecAllow = "hevc,h265,x265,HEVC,h.265,hevc_nvenc,yuv420p10le,hevcmain,hevc10,yuv420p"

---

Here is what its doing with a x265 file
[11:15:02] [INFO]::MAIN: Loading config from [F:\Scripts\nzbtomedia\autoProcessMedia.cfg]
[11:15:02] [INFO]::MAIN: Python v3.7 will reach end of life in 1083 days.
[11:15:02] [INFO]::MAIN: Checking database structure...
[11:15:02] [DEBUG]::MAIN: Checking Initial Schema database upgrade
[11:15:02] [DEBUG]::MAIN: InitialSchema upgrade not required
[11:15:02] [INFO]::MAIN: Checking if source needs an update
[11:15:03] [DEBUG]::MAIN: cur_commit = 8a0ba932aea0c1aff47de8c0ba3a1ee62c527d00 % (newest_commit)= 8a0ba932aea0c1aff47de8c0ba3a1ee62c527d00, num_commits_behind = 0
[11:15:03] [INFO]::MAIN: No update needed
[11:15:03] [INFO]::MAIN: nzbToMedia Version:8a0ba932aea0c1aff47de8c0ba3a1ee62c527d00 Branch:dev (Windows 10)
[11:15:03] [INFO]::MAIN: #########################################################
[11:15:03] [INFO]::MAIN: ## ..::[nzbToMedia.py]::.. ##
[11:15:03] [INFO]::MAIN: #########################################################
[11:15:03] [DEBUG]::MAIN: Options passed into nzbToMedia: ['\\Scripts\\nzbtomedia\\nzbToNzbDrone.py']
[11:15:03] [WARNING]::MAIN: Invalid number of arguments received from client, Switching to manual run mode ...
[11:15:03] [INFO]::MAIN: Searching F:\NZB\TV for mediafiles to post-process ...
[11:15:03] [DEBUG]::MAIN: Found file #blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv in root directory F:\NZB\TV.
[11:15:03] [INFO]::COPYLINK: MEDIAFILE: [b'#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv']
[11:15:03] [INFO]::COPYLINK: SOURCE FOLDER: [F:\NZB\TV]
[11:15:03] [INFO]::COPYLINK: TARGET FOLDER: [b'F:\\NZB\\TV\\blackAF']
[11:15:03] [INFO]::COPYLINK: Moving SOURCE MEDIAFILE -> TARGET FOLDER
[11:15:03] [DEBUG]::MAIN: Found file #blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv in root directory F:\NZB\TV.
[11:15:03] [INFO]::COPYLINK: MEDIAFILE: [b'#blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv']
[11:15:03] [INFO]::COPYLINK: SOURCE FOLDER: [F:\NZB\TV]
[11:15:03] [INFO]::COPYLINK: TARGET FOLDER: [b'F:\\NZB\\TV\\blackAF']
[11:15:03] [INFO]::COPYLINK: Moving SOURCE MEDIAFILE -> TARGET FOLDER
[11:15:03] [INFO]::MAIN: Starting manual run for NzbDrone:tv - Folder: F:\NZB\TV\blackAF
[11:15:03] [INFO]::MAIN: Checking database for download info for blackAF ...
[11:15:03] [INFO]::MAIN: Unable to locate download info for blackAF, continuing to try and process this release ...
[11:15:03] [INFO]::MAIN: Auto-detected SECTION:NzbDrone
[11:15:03] [DEBUG]::MAIN: Checking for archives to extract in directory: F:\NZB\TV\blackAF
[11:15:03] [INFO]::MAIN: Calling NzbDrone:tv to post-process:blackAF
[11:15:03] [DEBUG]::SERVER: Attempting to connect to server at http://localhost:9000
[11:15:03] [DEBUG]::SERVER: Server responded at http://localhost:9000
[11:15:03] [INFO]::MAIN: Attempting to verify tv fork
[11:15:03] [INFO]::MAIN: NzbDrone:tv fork set to default
[11:15:03] [INFO]::MAIN: FLATTEN: Flattening directory: F:\NZB\TV\blackAF
[11:15:03] [DEBUG]::MAIN: Checking for empty folders in:F:\NZB\TV\blackAF
[11:15:03] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffprobe.exe -v quiet -print_format json -show_format -show_streams -show_error F:\Scripts\nzbtomedia\tests\test.mp4
[11:15:03] [INFO]::TRANSCODER: Checking [#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv] for corruption, please stand by ...
[11:15:03] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffprobe.exe -v quiet -print_format json -show_format -show_streams -show_error F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv
[11:15:04] [INFO]::TRANSCODER: SUCCESS: [#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv] has no corruption.
[11:15:04] [INFO]::SUBTITLES: Attempting to download subtitles for F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv
[11:15:08] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffprobe.exe -v quiet -print_format json -show_format -show_streams -show_error F:\Scripts\nzbtomedia\tests\test.mp4
[11:15:08] [INFO]::TRANSCODER: Checking [#blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv] for corruption, please stand by ...
[11:15:08] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffprobe.exe -v quiet -print_format json -show_format -show_streams -show_error F:\NZB\TV\blackAF\#blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv
[11:15:08] [INFO]::TRANSCODER: SUCCESS: [#blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv] has no corruption.
[11:15:08] [INFO]::SUBTITLES: Attempting to download subtitles for F:\NZB\TV\blackAF\#blackAF - S01E03 - still. because of slavery [h265 EAC3 Atmos 8].mkv
[11:15:11] [INFO]::MAIN: Checking for files to be transcoded
[11:15:11] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffprobe.exe -v quiet -print_format json -show_format -show_streams -show_error F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv
[11:15:11] [INFO]::MAIN: Transcoding video: F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10]-transcoded.mkv
[11:15:12] [DEBUG]::MAIN: calling command: f:\Scripts\ffmpeg\ffmpeg.exe -loglevel warning -hwaccel auto -fflags +genpts -i F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10].mkv -map 0:0 -map 0:1 -map 0:2 -c:v hevc_nvenc -preset hp -crf 30 -c:a:0 copy -c:s copy F:\NZB\TV\blackAF\#blackAF - S01E01 - because of slavery [HEVC EAC3 Atmos 10]-transcoded.mkv

**Comment by clinton-hall** [comment]
Unfortunately I still have a big list of improvements to make to the transcoder to allow better handling of these things...

Currently, the VideoCodecAllow is lower in precedence to
    outputVideoPreset = 
    outputVideoResolution = 
    outputVideoFramerate = 
    outputVideoBitrate = 
    outputVideoCRF = 

So it appears from the log, you have specified:
    outputVideoPreset = hp
    outputVideoCRF = 30

If you set these both to blank, then it should just COPY any video streams that are in VideoCodecAllow... in this case it would ve a very quick Transcode where it has 1:1 mapping, with copying of existing streams... 
In an ideal world I also need to make it smarter and not transcode if it isn't actually changing anything!

In Short, The stream will be copied if the Codec is in VideoCodecAllow. But as soon as you specify any specific Output requirements, the transcoder will convert the video

Notes:
 outputVideoResolution does actually check the current resolution and only applies if the current resolution is higher... i.e. it will down-scale
outputVideoFramerate does check for current frame rate and only applies if the current framerate is more than 0.1% off...

**Comment by pcxnet** [comment]
Thanks, also i saw in the raw coding a reference to autoprocessmovie.cfg and autoprocesstv.cfg

Does it use those in place of autoprocessmedia.cfg if they exist? 

I couldn't see any docs on it.

Thanks

**Comment by clinton-hall** [comment]
autoprocessmovie.cfg and autoprocesstv.cfg are legacy from when this changed from `sabToCouchPotato` (which used autoprocessmovie.cfg) and I set it up to import that and import autoprocesstv.cfg from `sabToSickBeard`...

So if autoprocessmovie.cfg and autoprocesstv.cfg exist, this script will migrate/import those settings into autoProcessMedia.cfg... But I strong suggest NOT doing that... I haven't looked at that for quite some time...

**Comment by pcxnet** [comment]
No problem.  Thanks for the help :)
